linux/ibus-keyman/src/engine.c

```diff
@@ -210,14 +210,30 @@ ibus_keyman_engine_constructor (GType                   type,
 
     keyman->kb_name = NULL;
     keyman->ldmlfile = NULL;
-    gchar *kmx_file = g_path_get_basename(engine_name);
+    gchar **split_name = g_strsplit(engine_name, ":", 2);
+    if (split_name[0] == NULL)
+    {
+        abs_kmx_path = NULL;
```

Comment:
What will happen if `abs_kmx_path` is `NULL`? Won't that cause errors further down (e.g. line 229)?

Reply:
aye, a very unlikely state, but changed it to abort construction